ADALiOS/ADALiOS/ADAuthenticationRequest+AcquireToken.m

```diff
@@ -335,7 +354,7 @@ - (void) validatedAcquireTokenByRefreshToken:(NSString*)refreshToken
           additionalHeaders:nil
                  completion:^(NSDictionary *response)
      {
-         ADTokenCacheStoreItem* resultItem = (cacheItem) ? cacheItem : [ADTokenCacheStoreItem new];
+         ADTokenCacheStoreItem* resultItem = (cacheItem) ? [ADTokenCacheStoreItem initWithStoreItem:cacheItem] : [ADTokenCacheStoreItem new];
```

Comment:
It just means you found even more bugs! :)

The logic in this bit of code is a tad convoluted and the error handling is nowhere near as explicit as it should be. I would recommend instead checking for a AUTH_NON_PROTOCOL_ERROR in the response dictionary and sending that along to the completionBlock instead of executing any of the code in this block.